src/lib.rs

```diff
@@ -2249,4 +2298,111 @@ mod tests {
 			}
 		}
 	}
+
+	#[cfg(feature = "x509-parser")]
+	mod test_ip_address_from_octets {
+		use std::net::IpAddr;
+		use super::super::ip_addr_from_octets;
+		use super::super::RcgenError;
+
+		#[test]
+		fn ipv4() {
+			let actual = ip_addr_from_octets(&[10, 20, 30, 40])
+				.unwrap();
+
+			let expected = IpAddr::from([10, 20, 30, 40]);
+
+			assert_eq!(expected, actual)
+		}
+
+		#[test]
+		fn ipv6() {
+			let actual = ip_addr_from_octets(
+				&[
+					1,
```

Comment:
This formatting is not great at all. Can it be on one line? Also if you have the same slice then it might make sense to just put it into a variable? Lastly, if it's increasing numbers you could probably do it like in the `too_many` test, via collecting a range.

Reply:
Creating the input with a range and collecting it to a vector won't compile, because `IpAddr::from` requires a fixed-length array.

I can deduplicate the octets though and make it one line.